File: Programmers/가장 큰 수.js
function solution(numbers) {
    var strArr = numbers.map(number => number.toString())
    var maxLeng = strArr.map(char => char.length).reduce((a, b) => Math.max(a, b))
    var sortedArr = strArr.map(function(number) {
        var idx = strArr.indexOf(number)
        if (number.length < maxLeng) {
            for (let i=0; i<maxLeng-number.length+1; i++) {
                number += number[0];
            }
        }
        return [number, idx]
    })
    sortedArr = sortedArr.sort((a, b) => b[0] - a[0])
    var answer = [];
    for (let j=0; j<sortedArr.length; j++) {
        answer.push(strArr[sortedArr[j][1]])
    }
    if (answer[0] == 0) {
        return '0'
    } else return answer.join('')
}

// 시간 초과로 실패

File: Programmers/가장 큰 수.py
def solution(numbers):
	str_arr = list(map(str, numbers))
	max_length = 0
	for num in str_arr:
		if len(num) > max_length:
			max_length = len(num)
	sorted_arr = []
	for idx, num in enumerate(str_arr):
		if len(num) < max_length:
			num += num[0] * (max_length - len(num))
		sorted_arr += [[int(num), idx]]
	sorted_arr.sort(key=lambda x: -x[0])
	answer = []
	for num in sorted_arr:
		answer += str_arr[num[1]]
	return '0' if answer[0] == '0' else ''.join(answer)
